include/dxc/Support/HLSLOptions.td

```diff
@@ -172,6 +172,10 @@ def fdisable_loc_tracking : Flag<["-"], "fdisable-loc-tracking">,
   Group<hlslcomp_Group>, Flags<[CoreOption]>,
   HelpText<"Disable source location tracking in IR. This will break diagnostic generation for late validation. (Ignored if /Zi is passed)">;
 
+def fdisable_always_inline : Flag<["-"], "fdisable-always-inline">,
```

Comment:
Shall we add it as something like always_force_no_inline which always add no inline on all functions to expose more issues?

Reply:
I would prefer if it was added to our opt enable/disable system if that makes sense, like: `-opt-disable always-inline`

Reply:
It isn't really disabling a pass... it is changing the inlining policy (swapping inliners and the timing of the pass in the pipeline), as well as changing how we generate function declarations.

Reply:
> Shall we add it as something like always_force_no_inline which always add no inline on all functions to expose more issues?

Forcing adding `noinline` isn’t the goal here. The goal is to enable a compiler mode where profitable inlining occurs but we don’t force _unprofitable_ inlining the way that the `alwaysinline` attribute does.